### PR TITLE
Add DeviceOrientation event

### PR DESCRIPTION
**Pull request by felquis**
Event attached only to window object

### CODE REVIEW

File: src/Example.js [hunk 1]
```diff
@@ -8,16 +8,33 @@ export default class Example extends Component {
...
   }
 
   move(position) {
     this.setState(position)
   }
 
+  handleDeviceOrientation({gamma, alpha, beta, absolute}) {
+    this.setState({
+      deviceorientation: {
+        absolute: absolute,
```
Comment (MarcoWorms) [inline src/Example.js]: You could use the shorthand property syntax here :)

File: src/Example.js [hunk 2]
```diff
@@ -51,6 +68,15 @@ export default class Example extends Component {
           })
         }} />
 
+        <DeviceOrientation do={(event) => this.handleDeviceOrientation(event)} />
+
+        <div>
+          deviceorientation:
```
Comment (MarcoWorms) [inline src/Example.js]: weird line

File: src/Example.js (hunk 1 above)
Comment (MarcoWorms) [inline src/Example.js]: on a second thought you just destructed the object and then structured it again, why not just pass it through as an object?

Comment (felquis) [inline src/Example.js]: Cause the event itself has a lot of other properties, i just want these four for the demo

File: src/Example.js (hunk 2 above)
Comment (felquis) [inline src/Example.js]: Fixed

File: src/Example.js (hunk 1 above)
Comment (felquis) [inline src/Example.js]: Could you suggest an alternative pls?

Comment (MarcoWorms) [inline src/Example.js]:
```javascript
handleDeviceOrientation( deviceOrientation ) {
  this.setState({ deviceOrientation })
}
```

Comment (MarcoWorms) [inline src/Example.js]: just saw the problem with that, i'll come with a better alternative

Comment (MarcoWorms) [inline src/Example.js]: well, i think this is the shortest way:

```javascript 
handleDeviceOrientation(deviceOrientation = {gamma, alpha, beta, absolute}) {
  this.setState({ 
    deviceOrientation: { 
      gamma, 
      alpha, 
      beta, 
      absolute 
    } 
  })
}
```

Comment (felquis) [inline src/Example.js]: :+1:

File: src/react-event-components/DeviceOrientation.js [hunk 3]
```diff
@@ -0,0 +1,26 @@
+const { Component, PropTypes } = require('react')
+
+class DeviceOrientation extends Component {
```
Comment (derekstavis) [inline src/react-event-components/DeviceOrientation.js]: I think we could name this `DeviceOrientationEvent`, on par with `KeyEvent`

**Comment by felquis** [comment]
@MarcoWorms rebased and fixed!

**Comment by felquis** [comment]
@derekstavis @MarcoWorms rebased again, and added DeviceOrientation Example :)

**Comment by derekstavis** [comment]
It's missing a README example! 😛

**Comment by felquis** [comment]
@derekstavis added example to README! :)

**Pull request merged**